tests/serialized_test_data_generation/submit_job.sh

```diff
@@ -1,12 +1,37 @@
 #!/bin/bash
 
+# stop on all errors
+set -e
+
+# setup run directory
 RUNDIR=/rundir
+cp /FV3/fv3.exe $RUNDIR/fv3.exe
+
+# setup directory for serialized data
 TEST_DATA_DIR=$RUNDIR/test_data
+mkdir -p $TEST_DATA_DIR
+
+# setup environment
+
+# run the model (deduce MPI configuration from config file)
+cd $RUNDIR
 NUM_RANKS=`python ${RUNDIR}/count_ranks.py $RUNDIR/fv3config.yml`
 ulimit -s unlimited
-cp /FV3/fv3.exe $RUNDIR/fv3.exe
-mkdir -p $TEST_DATA_DIR
+echo ">>> Running model, inspect stdout.out / stderr.out in rundir for details"
+mpirun -l -np $NUM_RANKS $RUNDIR/fv3.exe 2>$RUNDIR/stderr.out 1>$RUNDIR/stdout.out
+echo ">>> Success"
+
+# copy artefacts to test_data directory
 cp $RUNDIR/input.nml $TEST_DATA_DIR/
 cp $RUNDIR/fortran_sha.txt $TEST_DATA_DIR/
-cd $RUNDIR
-mpirun -np $NUM_RANKS --allow-run-as-root --mca btl_vader_single_copy_mechanism none --oversubscribe $RUNDIR/fv3.exe
+cp $RUNDIR/logfile.*.out $TEST_DATA_DIR/
+cp $RUNDIR/std*.out $TEST_DATA_DIR
+env > $TEST_DATA_DIR/env.out
```

Comment:
what does this spit out?

Reply:
It echoes all environment variables defined at the time when the model was executed.